pkg/gornir/gornir.go

```diff
@@ -44,10 +49,15 @@ func (g *Gornir) RunS(title string, runner Runner, task Task) (chan *JobResult,
 // RunA will execute the task over the hosts in the inventory using the given runner.
 // This function doesn't block, the user can use the method Runnner.Wait instead.
 // It's also up to the user to ennsure the channel is closed
-func (g *Gornir) RunA(title string, runner Runner, task Task, results chan *JobResult) error {
+func (gr *Gornir) RunA(title string, runner Runner, task Task, results chan *JobResult) error {
 	err := runner.Run(
-		NewContext(context.Background(), title, g, g.Logger),
+		context.Background(), // TODO pass this?
 		task,
+		&TaskParameters{
+			Title:  title,
+			Gornir: gr,
+			Logger: gr.Logger,
```

Comment:
replace with something like gr.Logger.WithNewID() or similar that returns a new logger with an attached uid